listenbrainz_spark/request_consumer/request_consumer.py

```diff
@@ -58,34 +58,35 @@ def get_result(self, request):
             return None
 
 
-    def push_to_result_queue(self, result):
-        while True:
-            try:
-                self.result_channel.basic_publish(
-                    exchange=current_app.config['SPARK_RESULT_EXCHANGE'],
-                    routing_key='',
-                    body=json.dumps(result),
-                    properties=pika.BasicProperties(delivery_mode = 2,),
-                )
-                break
-            except pika.exceptions.ConnectionClosed:
-                self.connect_to_rabbitmq()
-                time.sleep(1)
-            except pika.exceptions.ChannelClosed:
-                self.result_channel = self.rabbitmq.channel()
-                self.result_channel.exchange_declare(exchange=current_app.config['SPARK_RESULT_EXCHANGE'], exchange_type='fanout')
+    def push_to_result_queue(self, messages):
+        for message in messages:
+            while True:
+                try:
+                    self.result_channel.basic_publish(
+                        exchange=current_app.config['SPARK_RESULT_EXCHANGE'],
+                        routing_key='',
+                        body=json.dumps(message),
```

Comment:
Hmm. now that we merged in the -labs repo we're using json and ujson -- we should make an effort to replace all json with ujson...

Reply:
This is a bit more complicated and we should take a look at dependencies a bit more holistically. I'm not gonna do it here, but open a ticket.